module/connection/ServerDialog.kt

```diff
@@ -175,31 +200,75 @@ object ServerDialog {
     }
 
     @Composable
-    private fun ConnectServer() = Dialog.Layout(
-        state = Service.driver.connectServerDialog,
-        title = Label.CONNECT_TO_TYPEDB,
-        width = WIDTH,
-        height = HEIGHT
-    ) {
-        Submission(state = state, modifier = Modifier.fillMaxSize(), showButtons = false) {
-            ServerFormField(state)
-            if (state.server == TYPEDB_CLOUD) {
-                ManageCloudAddressesButton(state = state, shouldFocus = Service.driver.isDisconnected)
-                UsernameFormField(state)
-                PasswordFormField(state)
-                TLSEnabledFormField(state)
-                if (state.tlsEnabled) CACertificateFormField(state = state, dialogWindow = window)
-            } else if (state.server == TYPEDB_CORE) {
-                CoreAddressFormField(state, shouldFocus = Service.driver.isDisconnected)
-            }
-            Spacer(Modifier.weight(1f))
-            Row(verticalAlignment = Alignment.Bottom) {
-                ServerConnectionStatus()
-                Spacer(modifier = Modifier.weight(1f))
-                when (Service.driver.status) {
-                    DISCONNECTED -> DisconnectedFormButtons(state)
-                    CONNECTING -> ConnectingFormButtons()
-                    CONNECTED -> ConnectedFormButtons(state)
+    private fun ColumnScope.AdvancedConfigToggleButtons(state: ConnectServerForm) {
+        TextButtonRow(
+            height = TOOLBAR_BUTTON_SIZE,
+            buttons = listOf(
+                TextButtonArg(
+                    text = Label.CONNECTION_URI,
+                    onClick = { state.advancedConfigOpen = false; updateHeight() },
+                    color = { toggleButtonColor(!state.advancedConfigOpen) },
+                ),
+                TextButtonArg(
+                    text = Label.ADVANCED_CONFIG,
+                    onClick = { state.advancedConfigOpen = true; updateHeight() },
+                    color = { toggleButtonColor(state.advancedConfigOpen) },
+                )
+            ),
+            modifier = Modifier.align(Alignment.CenterHorizontally)
+        )
+    }
+
+    private fun updateHeight() {
+        val currentHeight = dialogState!!.size.height
+        dialogState!!.size = DpSize(
+            width = dialogState!!.size.width,
+            height = if (state.advancedConfigOpen && currentHeight < ADVANCED_HEIGHT) ADVANCED_HEIGHT
+            else if (!state.advancedConfigOpen && currentHeight == ADVANCED_HEIGHT) SIMPLE_HEIGHT
+            else currentHeight
+        )
+    }
+
+    @Composable
+    private fun ConnectServer() {
+        if (this.dialogState == null) {
+            val dialogState = rememberDialogState(
+                position = Aligned(Alignment.Center),
+                size = DpSize(WIDTH, if (state.advancedConfigOpen) ADVANCED_HEIGHT else SIMPLE_HEIGHT)
+            )
+            this.dialogState = dialogState
+        }
```

Comment:
`rememberDialogState` must be called from an `@Composable` function, so we initialise `this.dialogState` here rather than inlined with its definition